### PR TITLE
build(deps): fix unchecked lock and update deps

### PR DESCRIPTION
**Pull request by j-mendez**
1. build(deps): update cssparser@0.29.6 and selector@0.24.0 (mem reduction, perf increase, and css3 syntax fixes )
1. fix missing Cargo.lock file
1. add lazy evaluation classes

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "scraper"
-version = "0.14.0"
+version = "0.14.1"
```
Comment (cfvescovo) [inline Cargo.toml:3]: We never edit version number in PRs. Version numbers are bumped when we cut and publish a new release.

File: src/node.rs [hunk 2]
```diff
@@ -247,7 +247,7 @@ impl Element {
         let classes: HashSet<LocalName> = attrs
             .iter()
             .find(|a| a.name.local.deref() == "class")
-            .map_or(HashSet::new(), |a| {
+            .map_or_else(|| HashSet::new(), |a| {
```
Comment (cfvescovo) [inline src/node.rs:250]: Correct me if I am wrong but this doesn't seem necessary.

**Review by cfvescovo — CHANGES_REQUESTED** [review]

**Comment by j-mendez** [comment]
@cfvescovo @cfvescovo `.or` is evaluated regardless if it is true. For this situation we do not need to create a new hashset eagerly and can move it lazily. 

We can use or for this block in the `element_ref` and it makes sense since we want this to be fast during this execution. 

```rust
    fn is_root(&self) -> bool {
        self.parent()
            .map_or_else(|| false, |parent| parent.value().is_document())
    }
```

**Comment by cfvescovo** [comment]
I know but Rust's collections generally don't allocate on initialisation, so an eager String::new() or HashMap::new() doesn't really matter

**Comment by cfvescovo** [comment]
However I see your point so I removed my comment

**Comment by j-mendez** [comment]
> I know but Rust's collections generally don't allocate on initialisation, so an eager String::new() or HashMap::new() doesn't really matter

It does for `.map_or`

**Comment by cfvescovo** [comment]
I was referencing this: https://stackoverflow.com/a/73785382

**Comment by cfvescovo** [comment]
> > I know but Rust's collections generally don't allocate on initialisation, so an eager String::new() or HashMap::new() doesn't really matter
> 
> We want to leave the stack alone for this and should be fine to move it lazy. I can revert the change to keep it just with the build deps.

Keep your changes, I see your point. It won't make much of a difference but ok.

**Comment by j-mendez** [comment]
@cfvescovo thank you for the fast replies - fixing the format issues now.

**Comment by cfvescovo** [comment]
Cargo fmt is complaining, rerun the formatter

**Comment by cfvescovo** [comment]
@j-mendez thank you!

**Pull request merged**